File: API/API/Features/v1/EmployeeArea/EmployeeResponse.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace CLERP.API.Features.v1.EmployeeArea
{
    /// <summary>
    /// Default response dto for an employee
    /// </summary>
    public class EmployeeResponse
    {
        [JsonProperty("id")]
        public Guid Guid { get; set; }

        [JsonProperty("firstname")]
        public string Firstname { get; set; }

        [JsonProperty("lastname")]
        public string Lastname { get; set; }

        [JsonProperty("birthday")]
        public DateTime Birthday { get; set; }

        [JsonProperty("email")]
        public string Email { get; set; }

        [JsonProperty("phoneNumber")]
        public string PhoneNumber { get; set; }

        [JsonProperty("username")]
        public string Username { get; set; }

        [JsonProperty("department")]
        public EmployeeDepartmentResponse Department { get; set; }

        [JsonProperty("roles")]
        public IEnumerable<EmployeeRoleResponse> Roles { get; set; }
    }
}

File: Client/src/app/_generated/models/employee-response.ts
/* tslint:disable */
import { EmployeeDepartmentResponse } from './employee-department-response';
import { EmployeeRoleResponse } from './employee-role-response';

/**
 * Default response dto for an employee
 */
export interface EmployeeResponse {
  id?: string;
  firstname?: string;
  lastname?: string;
  birthday?: string;
  email?: string;
  phoneNumber?: string;
  username?: string;
  department?: EmployeeDepartmentResponse;
  roles?: Array<EmployeeRoleResponse>;
}
